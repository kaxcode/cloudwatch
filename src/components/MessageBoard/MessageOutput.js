import React from 'react';
import { func, string, bool } from 'prop-types';

const MessageOutput = props => {
  if (!props.showMessage) {
    return null;
  }
  return (
    <div className="message-output">
      <h1>{props.message}</h1>
      <div onClick={props.handleDismiss} className="message-output--cursor">
        X
      </div>
    </div>
  );
};

MessageOutput.propTypes = {
  showMessage: bool.isRequired,
  handleDismiss: func.isRequired,
  message: string.isRequired
};

export default MessageOutput;
