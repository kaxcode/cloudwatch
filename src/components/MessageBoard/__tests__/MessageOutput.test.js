import React from 'react';
import Enzyme from 'enzyme';
import MessageOutput from '../MessageOutput.js';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('MessageOutput', () => {
  it('does not render a message when :showMessage is false', () => {
    const subject = shallow(<MessageOutput showMessage={false} />);
    expect(toJson(subject)).toMatchSnapshot();
  });
  it('renders a message when :showMessage is true', () => {
    const subject = shallow(<MessageOutput showMessage />);
    expect(toJson(subject)).toMatchSnapshot();
  });
});
