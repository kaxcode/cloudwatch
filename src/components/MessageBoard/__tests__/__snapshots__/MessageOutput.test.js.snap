// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageOutput does not render a message when :showMessage is false 1`] = `""`;

exports[`MessageOutput renders a message when :showMessage is true 1`] = `
<div
  className="message-output"
>
  <h1 />
  <div
    className="message-output--cursor"
  >
    X
  </div>
</div>
`;
